1 Add database schema

MaxShelestiuk opened the pull request.

MaxShelestiuk committed:
Add Database Schema to README.md

MaxShelestiuk committed:
Upload Database Schema image

Anatolii-Ivanov commented on README.md:
```diff
@@ -77,3 +77,6 @@ This is a list of the required environment variables:
 **SA_PASSWORD** - MSSQL Server "SA" user password
+
+## SQL Server Database Schema
+![Database_Schema](Squirrel_DB_Schema.svg)
```
Either use Mermaid for it or find a way to update svg file and add to documentation)

ABertashDev replied:
DB schema updated with Mermaid

ABertashDev committed:
Update DB Schema with Mermaid chart

ABertashDev committed:
Delete DB Schema image

ABertashDev committed:
Fixed incorrect types of relationships between tables

alinaSielina reviewed (commented):
Looks good, but I have a few clarifications:
- UserSettings and ProjectSettings tables. I'm not entirely sure about the purpose of the SettingsUrl column. Given the relatively simple addition, I suggest we consider including these tables later, perhaps when we implement features such as the settings page. Their inclusion should not greatly affect the overall structure.
- The GroupsTags, Groups and UserGroups tables are redundant. Instead, we can update the UserProjects table by adding a Role(int) column and then later include the RoleEnum in our code.
- Branch table - I understand the IsActive column will be used for soft delete? We decided to implement a soft delete function, but maybe we should call it IsDeleted?
- Comments table - we will use comments not only for PRs, so I think we should now expand this table, and it could look like this for example:
   Id (int), 
   AuthorId (int), 
   Content (nvarchar(500)), 
   CreatedAt (datetime), 
   UpdatedAt (datetime), 
   CommentableId (int), 
   CommentableType (int)
   
- ProjectFiles table. I don't fully understand why this table refers to Users and Projects and not Commits. Or maybe you added this table for the README files? So I think we can use a Git-like structure:
   
![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/139565128/4c8b2462-6c9c-435a-8fa0-7db05d9263c5)

![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/139565128/e2c0623d-ec62-43d9-907e-67fd36be217c)

something like this:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/139565128/d77f09ec-81b0-47dc-80f2-44ae952c581e)

OlegKarapysh commented on README.md:
```diff
@@ -77,3 +77,182 @@ This is a list of the required environment variables:
+        nvarchar(25) FirstName
+        nvarchar(25) LastName
+        nvarchar(50) Email
+        nvarchar(25) Password
```
it is bad to store user's password explicitly, so maybe we should hash it with salt. In this case User entity should have 'Salt' attribute of nvarchar type

ABertashDev replied:
Thanks. Added Salt column and changed Password column to PasswordHash

ABertashDev committed:
Update Database schema after coach reviews

- Improved Commit data logic.
- Removed Groups and Settings tables.
- Added RoleEnum to UserProjects table.
- Modified Comments table to relate with multiple entities.
- Updated password storing (Salt & PasswordHash)
